server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java

```diff
@@ -226,63 +116,170 @@ public void testFullClusterRestoreManifestFilePointsToInvalidIndexMetadataPathTh
+    }
+
+    public void testRemoteStateRestoreLongRunning() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 1;
+        int dataNodeCount = shardCount * (replicaCount + 1);
+        int clusterManagerNodeCount = 3;
+
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
 
-        // Step - 5 Trigger full cluster restore and validate fails
-        PlainActionFuture<RestoreRemoteStoreResponse> future = PlainActionFuture.newFuture();
-        restoreAndValidateFails(prevClusterUUID, future, IllegalStateException.class, "asdsa");
+        for (int i = 0; i < 100; i++) {
+            logger.info("Current iteration is: {}", i);
+            performOperation(i);
+            validateCurrentMetadata();
+            String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+            Metadata prevMetadata = clusterService().state().metadata();
+            resetCluster(dataNodeCount, clusterManagerNodeCount);
+            String newClusterUUID = clusterService().state().metadata().clusterUUID();
+            assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
+
+            validateMetadataAfterRestore(prevMetadata);
+        }
     }
 
-    private void reduceShardLimits(int maxShardsPerNode, int maxShardsPerCluster) {
-        // Step 3 - Reduce shard limits to hit shard limit with less no of shards
-        try {
-            client().admin()
-                .cluster()
-                .updateSettings(
-                    new ClusterUpdateSettingsRequest().transientSettings(
-                        Settings.builder()
-                            .put(SETTING_CLUSTER_MAX_SHARDS_PER_NODE.getKey(), maxShardsPerNode)
-                            .put(SETTING_MAX_SHARDS_PER_CLUSTER_KEY, maxShardsPerCluster)
-                    )
-                )
-                .get();
-        } catch (InterruptedException | ExecutionException e) {
-            throw new RuntimeException(e);
+    public void testRemoteStateFullRestart() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 1;
+        int dataNodeCount = shardCount * (replicaCount + 1);
+        int clusterManagerNodeCount = 3;
+
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
+        String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+        internalCluster().fullRestart();
+        ensureGreen(INDEX_NAME);
+        String newClusterUUID = clusterService().state().metadata().clusterUUID();
+        assert Objects.equals(newClusterUUID, prevClusterUUID) : "Full restart not successful. cluster uuid has changed";
+        validateCurrentMetadata();
+        verifyRestoredData(indexStats, INDEX_NAME);
```

Comment:
you mean restored from disk right? how are we asserting it is not restoring from remote for only restart usecases.

Reply:
Yeah, we are not. Let me add an assertion for this. Thanks for catching this.